astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -399,19 +400,7 @@ def test_astype(self):
                 (10 * u.radian, None),
                 {},
                 np.rad2deg(np.arange(10, dtype=float) * ARCSEC_PER_DEGREE),
-                id="pos: stop, followed by 1 None",
-            ),
-            pytest.param(
-                (10 * u.radian, None, None),
-                {},
-                np.rad2deg(np.arange(10, dtype=float) * ARCSEC_PER_DEGREE),
-                id="pos: stop, followed by 2 None",
-            ),
-            pytest.param(
-                (10 * u.radian, None, None, None),
-                {},
-                np.rad2deg(np.arange(10, dtype=float) * ARCSEC_PER_DEGREE),
-                id="pos: stop, followed by 3 None",
```

Comment:
I could have salvaged these tests either by using `ArangeDefaults.STEP`, but then the tests wouldn't represent valid and supported use cases anymore.

Although... trying to explain this fully made me realize there should be another, preferable avenue. I'll give it a try now!